src/registry/parseRegistry.ts

```diff
@@ -1,21 +1,44 @@
-// Copyright 2023 Parity Technologies (UK) Ltd.
+// Copyright 2023-2024 Parity Technologies (UK) Ltd.
 
 import { ASSET_HUB_CHAIN_ID } from '../consts';
-import type { AssetTransferApiOpts } from '../types';
-import type { ChainInfoRegistry } from './types';
+import type { AssetTransferApiInjectedOpts } from '../types';
+import type { ChainInfo, ChainInfoRegistry, InjectedChainInfo } from './types';
 
-export const parseRegistry = (registry: ChainInfoRegistry, assetsOpts: AssetTransferApiOpts): ChainInfoRegistry => {
+const updateRegistry = (injectedChain: InjectedChainInfo, registry: ChainInfoRegistry, registryChain: string) => {
+	for (const key of Object.keys(injectedChain)) {
+		const info = registry[registryChain] as unknown as ChainInfo;
+		if (info[key] !== undefined) {
+			Object.assign(info[key], injectedChain[key]);
+		} else {
+			for (const property of Object.keys(info[0])) {
+				if (injectedChain[key][property] === undefined) {
+					if (property === 'specName' || property === 'tokens') {
```

Comment:
Why are we singling out specName, and tokens here, what about the rest of the keys?

Reply:
Because in this case we are adding a chain to the registry, and it must have `specName` and `tokens`, the other fields are autofilled with `{}` as per line 18.

Reply:
I added a new test where you can see this at work